### PR TITLE
fix: ADB.pidof integration issue with grep and CRLF edge case

### LINKED ISSUE #924: Regex for Android bundle id causing failures
Using Mac - High Sierra

#### Description
Line 113 of src/devices/android/ADB.js is: 
`const bundleIdRegex = pipeCommands.escape.inQuotedRegexp(bundleId) + '[ ]*$';`

I'm not sure what the `'[ ]*$'` is for, or maybe I'm building my application incorrectly.  I'm only able to get detox 9.0.1 to work correctly when I delete the `+ '[ ]*$'` from that file.  I know this is somewhere between and an issue and a question, but can someone clear this up for me.

Sample error:
detox[9144] ERROR: [exec.js/EXEC_FAIL, #19] 
detox[9144] ERROR: [exec.js/EXEC_FAIL, #20] "/Users/me/Library/Android/sdk/platform-tools/adb -s emulator-5554 shell ps | grep "com\.myapp\.myappportal[ ]*$"" failed with code = 1, stdout and stderr:

### PR DESCRIPTION
**Pull request by noomorph**
Resolves #924

### CODE REVIEW

File: detox/src/devices/android/ADB.js [hunk 1]
```diff
@@ -6,8 +6,8 @@ const EmulatorTelnet = require('./EmulatorTelnet');
 const Environment = require('../../utils/environment');
 
 class ADB {
-  constructor() {
-    this.adbBin = path.join(Environment.getAndroidSDKPath(), 'platform-tools', 'adb');
+  constructor({ adbBin } = {}) {
```
Comment (rotemmiz) [inline detox/src/devices/android/ADB.js]: where do you pass adbBin to the constructor? What's the usecase for this ?

Comment (noomorph) [reply detox/src/devices/android/ADB.js]: @rotemmiz , was my explanation sufficient? Can we merge this PR?

**Comment by noomorph** [comment]
@tomatobrown, technically, this PR is ready, but two things did not happen:

1) it was not approved by maintainers

And honestly, I'm still confused about what's happening in your environment with CRLF line endings from Android Emulator on Mac. It is not supposed to happen at all. I think Rotem is hesitant to approve exactly because of this concern.

File: detox/src/devices/android/ADB.integration.test.js [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+        mWakeUpWhenPluggedOrUnpluggedInTheaterModeConfig=false
+        mUserActivityTimeoutOverrideFromWindowManager=${mUserActivityTimeoutOverrideFromWindowManager}
+        mUserInactiveOverrideFromWindowManager=false
+      `;
+
+      await fs.writeFile(tempPath, contents);
+      await adb.unlockScreen(deviceId);
+    }
+
+    describe('when unlocking an awake and unlocked device', function() {
```
Comment (noomorph) [inline detox/src/devices/android/ADB.integration.test.js]: Dumpsys power might also emit CRLF output, and this means that we are going into a wrong direction.

**Comment by noomorph** [comment]
I have erased the previous implementation in favor of a simpler one, using `adb exec-out` instead of `adb shell`. It feels right and if it proves to be working across all major platforms, I'll be happy to merge it soon.

**Comment by noomorph** [comment]
Unfortunately, `exec-out` had issues of its own, with lower API levels, with handling of `ps` command (the results were different from `adb shell` on some of Android versions). So, I'm trying now a third approach - to run grep inside `adb shell` itself and replace CRLF with LF for every stdout string we get.

Still on it.

**Comment by noomorph** [comment]
Guys, sorry for the delay. I just have finished the other unrelated but highly important task, so I finally can start sorting out the issues one by one. ETA - tomorrow I make a one more bugfix to PR and merge it

**Pull request merged**